electronics_lib/JLcTable.py

```diff
@@ -0,0 +1,38 @@
+import re
+from .PartsTable import *
+import numpy as np
+class JLcTable(LazyTable):
+  """Shared base class for JLCPCB product tables that contains common row definitions."""
+
+  DISTRIBUTER_PART_NUMBER = 'LCSC PART'
+  MANUFACTURER = 'Manufacturer'
+  PART_NUMBER = 'MFR.Part'
+  DATASHEETS = 'Datasheet'
+  DESCRIPTION = 'Description'
+  COST = PartsTableColumn(float)
+
+  @classmethod
+  def _parse_jlcpcb_common(cls, row: PartsTableRow) -> Dict[PartsTableColumn, Any]:
+    """Returns a dict with the cost row, or errors out with KeyError."""
+
+    #extracts out all the available prices for a given part by order quantity
+    price_list = re.findall(":(\d+\.\d*),", row['Price'])
+    float_array = [float(x) for x in price_list]
+
+    return {
+      #chooses the highest price, which is for the lowest quantity
+      cls.COST: float(max(float_array)),
+    }
+
+  @staticmethod
+  def parse(discription, regex_dictionary):
```

Comment:
Some short documentation on this method would be useful, including a short definition and example.